### DISCUSSION TITLE
Deformation issue

### DISCUSSION THREAD

**Discussion by jin0123456** [idaholab/moose#25538] [Q&A Modules: Solid mechanics] [answered]
Hello,
I want to know in MOOSE how to output the deformation displacement caused by a single strain model like creep? I hope someone can answer my question. Thanks.

**Comment by GiudGiud** [comment]
Hello

look for the variable disp_x in your output. 
Are you wanting to see the field or sample some values?

Guillaume

**Reply by jin0123456** [reply]
Like if there is a deformation caused by thermal expansion and creep, How do I know how much deformation is caused by thermal expansion or creep alone?

**Reply by GiudGiud** [reply]
Those are probably stored as strains. 
Use show_material_properties in the Debug block to see if they are. And if so you can output material properties with an option in the Outputs block

**Reply by jiangwen84** [reply]
You can only output strains separately. If you want, maybe you can find a way to integrate the strain and get the displacements from the strain. I do not believe we have that capability yet.

**Reply by jin0123456** [reply]
That's what I was thinking，thanks.

**Reply by hugary1995** [reply]
Maybe we should back up a little bit first: Do you want some measure of deformation that corresponds to the _entire_ body, or at every point in the domain? In other words, do you need a single number or some field outputs?

**Reply by jin0123456** [reply]
At every point in the domain I suppose.

**Reply by jin0123456** [reply]
Or a specific point on the boundary.

**Reply by hugary1995** [reply]
That you won't be able to tell directly, and reason is simple: displacements are macroscopic observables and you can't decompose them according to constitutive relations.

I second @jiangwen84 's suggestion. If you really want to decompose some global measure you'll have to get the global measure by homogenizing your quantities of interest in the first place. For example, integrate strain_xx to get total displacement in the x direction, and since strain = elastic strain + creep strain + thermal strain, you can then integrate each of them to get their respective contributions.

**Reply by jin0123456** [reply, marked as answer]
I see，thanks.
